spark/src/main/scala/org/apache/comet/serde/QueryPlanSerde.scala

```diff
@@ -59,12 +59,13 @@ object QueryPlanSerde extends Logging with ShimQueryPlanSerde with CometExprShim
     logWarning(s"Comet native execution is disabled due to: $reason")
   }
 
-  def supportedDataType(dt: DataType): Boolean = dt match {
+  def supportedDataType(dt: DataType, allowComplex: Boolean = false): Boolean = dt match {
```

Comment:
nit: Can we consider renaming this to `allowStruct` to make it explicit that this is only for structs (and not maps and arrays).

Reply:
renamed